systemtest/template_test.go

```diff
@@ -0,0 +1,194 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package systemtest_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/apm-server/systemtest"
+	"github.com/elastic/apm-server/systemtest/apmservertest"
+	"github.com/elastic/apm-server/systemtest/estest"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+)
+
+func TestIndexTemplateKeywords(t *testing.T) {
```

Comment:
FYI - I believe the `ignore_above` in the template is set within the libbeat code, and was not the main focus of the `1024` limitation test. 
IMO the important part is to test that the limit is set on the Intake API for fields that are indexed. But as you pointed out in the description, the Intake API and the indexed doc structure diverged quite a bit, so there would be some manual mapping required (which I agree we should get rid of). So in conclusion, I think it is fine to not test this on the Intake API anymore, but we might also open a discussion then if we even want to set the limit on the Intake API, or just align with the index setting - allow indexing a value > `1024`, but ignoring everything above the limit. WDYT?

Reply:
> I believe the ignore_above in the template is set within the libbeat code

Ah thanks, I didn't realise there was a default set by libbeat. In that case, let's remove this test and also remove ignore_above (where it's 1024) from all of our fields.yml definitions?

> but we might also open a discussion then if we even want to set the limit on the Intake API, or just align with the index setting - allow indexing a value > 1024, but ignoring everything above the limit. WDYT?

My gut feeling is that we should keep the limit as it is (AFAIK we haven't had any complaints?), but worth raising with the agent devs.

Reply:
Verified that by default `ignore_above: 1024` is set in [libbeat](https://github.com/elastic/beats/blob/master/libbeat/template/processor.go#L265-L271) for `keywords`.

Agreed that we can remove the limits from the `fields.yml` files (maybe in a follow up PR).